tests/cli-misc.rs

```diff
@@ -8,7 +8,6 @@ use crate::mock::clitools::{
     expect_ok_eq, expect_ok_ex, expect_stderr_ok, expect_stdout_ok, run, set_current_dist_date,
     this_host_triple, Config, Scenario,
 };
-use rustup::errors::TOOLSTATE_MSG;
```

Comment:
Just wondering about feedback. It makes sense that the old toolstate message isn't appropriate anymore, but is there a message that says something like: "Cannot update nightly channel because no newer nightlies were found that have all the required components: x, y z" and another like "Your nighty has been updated to version xxxx-xx-xx because more recent versions don't have the required components: x, y, z"?

Reply:
So, you do get this info-level message:
https://github.com/rust-lang/rustup.rs/blob/e2da58a59bdf34b68fa7074d4c42f11cbc5a22f9/src/dist/notifications.rs#L164-L168
as rustup searches backwards in history, though maybe we want an additional message saying which nightly was chosen in the end? Although that said, that _should_ be clear from the final install message.

As for "Cannot update nightly channel because no newer nightlies were found that have all the required components: x, y z", the only way that could happen is if the nightly you are already on is the newest one that has all components. In that case, rustup will give the same message it currently gives if you try to update and there are no updates available. The info messages from above should make it clear why any newer nightlies weren't downloaded.

So, with that all said, do you still think it's necessary to add or change any messages?

Also, to be clear that one removal was due to a previously failing test now succeeding since an older nightly had the required component :)

Reply:
I didn't see that info when I looked through the diff, so I suppose that probably has enough information to avoid confusion about what's happening.  It's maybe not the most elegant way of reporting, but that's just cosmetics. It could generate quite some output if you haven't updated in say a month. So I suppose it's balancing having a polished message against the hurdle of implementing it, which I can't really estimate how much work it would be. 

If rustup prints the version actually installed, than I don't think we need an extra message for that as long as there is something that hints at versions being skipped.

To summarize, I can definitely live with this and there are tons of not-cosmetic issues that are more worthwhile putting effort in.

Thanks alot for taking this on anyway, it will be a major quality of ux upgrade in any case.

Reply:
actually, probably it's rare to see components fail for more than 2 weeks, so the amount of output is probably quite reasonable, even after not updating for a long time.

Reply:
Yeah, I was about to say, you will only see that message as many times as there are nightlies since your component set last built successfully on nightly, no matter how old your nightly is. I'm also fairly sure that rustup prints which nightly was actually installed at the end, so we should be good on that end.

I agree with you that better cosmetics could be nice here, though I _think_ this gives us at least all the info, even if it's not presented in the cleanest way. My inclination would be to merge this and then do cosmetic fixes in follow-up work (maybe related to #1501).